src/SIM/SIMbase.C

```diff
@@ -2071,3 +2037,50 @@ void SIMbase::registerDependency (const std::string& name, SIMdependency* sim,
   this->registerDependency(sim,name,nvc,myModel,
                            this->getMADOF(basis,nvc).data());
 }
+
+
+void SIMbase::dumpEqSys(bool both)
+{
+  // Dump system matrix to file, if requested
+  for (DumpData& dmp : lhsDump)
+    if (dmp.doDump()) {
+      IFEM::cout <<"\nDumping system matrix to file "<< dmp.fname << std::endl;
+      std::ofstream os(dmp.fname.c_str(),
+                       dmp.step.size() == 1 ? std::ios::out : std::ios::app);
+      os << std::setprecision(17);
+      double old_tol = utl::zero_print_tol;
+      utl::zero_print_tol = dmp.eps;
+      SystemMatrix* M = myEqSys->getMatrix(0);
+      char matName[16];
+      if (dmp.step.size() == 1)
+        strcpy(matName,"A");
+      else
+        sprintf(matName,"A%d",dmp.count);
+      for (int i = 0; M; M = myEqSys->getMatrix(++i), ++matName[0])
+        M->dump(os,dmp.format,matName); // label matrices as A,B,C,...
+      utl::zero_print_tol = old_tol;
+    }
+
+  if (!both)
+    return;
```

Comment:
Do you actually need this? In case of eigenvalue solver there are usually no right-hand-side vectors, so maybe you can use `if (myEqSys->getNoRHS() =0) return;`instead?

Reply:
indeed, changed.